src/coreclr/vm/proftoeeinterfaceimpl.cpp

```diff
@@ -6038,15 +6039,13 @@ HRESULT ProfToEEInterfaceImpl::GetFunctionInfo2(FunctionID funcId,
         else
         {
             //
-            // We could not get any class information.
+            // Though missing frameInfo may not lead to an exact typeHandle match for the method's class,
```

Comment:
A bit of a nit, but I don't like leaving an empty else here. I think the comment explaining the change makes sense though

Reply:
We could replace the entire if(exactMatch) { } else if(...) { } else { } with an unconditional assignment of classId = TypeHandleToClassId(specificClass). The behavior should be identical to what you have here with fewer unneeded conditional checks.

Reply:
I mainly wanted to preserve the noted scenarios for when classID would be an exact match. I originally was opting for the unconditional assignment but saw that `specificClass` could be returned as null should this code path be hit https://github.com/dotnet/runtime/blob/edf1c6681a1262b3ef321572e106bb72553baf14/src/coreclr/vm/generics.cpp#L919-L920

Ill note the scenarios where the class should be an exact match and change it to 
```cpp
if (specificClass != NULL)
{
    classId = TypeHandleToClassId(specificClass);
}
```

Reply:
Yeah I remember looking at that and it wasn't clear to me how it would ever return NULL unless the genericArgsToken was both not NULL and not correct. Maybe out of caution we should change that code to do this:
```C++
MethodTable* pDeclaringMT = pMD->GetExactDeclaringType(dac_cast<PTR_MethodTable>(pExactGenericArgsToken));
if(pDeclaringMT != NULL)
{
    pMT = pDeclaringMT;
    retVal = TRUE;
}
```

@davidwrighton - that seem reasonable?

Reply:
Opting to keep the Null check in GetFunctionInfo2 instead of enforcing GetExactInstantiationsOfMethodAndItsClassFromCallInformation to never set pSpecificClass to NULL.